### PR TITLE
AAE-47025 Add check-pr-description composite action

### PR DESCRIPTION
**Pull request by GioLauria**
### Solution

Adds a new composite action `check-pr-description` that fails a pull request when its description is missing, too short, or **only a ticket/URL reference**.

**How it measures.** After stripping PR-template HTML comments, it removes URLs, autolinks, Markdown link targets and Jira ticket keys (e.g. `AAE-1234`), leaving the "meaningful" prose. That prose must satisfy **both**:
- `min-chars` (default `15`) — minimum characters, and
- `min-words` (default `3`) — minimum words (guards against single-token padding like `aaaaaaaaaaaaaaa`).

A body that is empty or only a ticket link/URL fails with a targeted `::error`.

**Automated PRs are skipped two ways** so they never need a description:
- **Author** — `*[bot]` plus a configurable `skip-authors` glob list.
- **Head branch** — a configurable `skip-branches` glob list (matched against `github.head_ref`), which also catches automation running under a normal service-account login. The default covers Dependabot, Renovate, updatecli, Flux image-update, release-please, changesets, Snyk, Mend/WhiteSource, propagation (`pr-*`) and generic `automated-*` / `automation/*` branches.

Draft PRs are skipped too. Both thresholds are validated as non-negative integers, and the `skip-authors`/`skip-branches` loops run with pathname expansion disabled. No repository checkout is required.

This is the **master** for parent [AAE-47024](https://hyland.atlassian.net/browse/AAE-47024); consumer repos reference it via a SHA-pinned `pull_request` workflow. It supersedes the earlier decentralized copy-paste approach (and the out-of-scope `automate-build-tools` pilot, now closed).

### Documentation

`docs/README.md` updated with a `### check-pr-description` section (usage snippet, inputs, author/branch skipping and a SHA-pinning note) plus a TOC entry. `check_readme.sh` passes.

### Tests

`test.yml` exercises the action in the `test` job: happy path, asserted failure on a too-short body, asserted failure on a **ticket-link-only** body, bot-author skip, and **automated-branch skip**. Optional `pr-body`/`pr-author`/`pr-draft`/`pr-branch` inputs make it testable off live PR context. `pre-commit` (including actionlint) passes; all shell logic paths verified locally.

### Ticket

[AAE-47025](https://hyland.atlassian.net/browse/AAE-47025) (parent: [AAE-47024](https://hyland.atlassian.net/browse/AAE-47024))

[AAE-47024]: https://hyland.atlassian.net/browse/AAE-47024?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by GioLauria — COMMENTED** [review]
Thanks @copilot-pull-request-reviewer — all four points addressed in `eefe39aa`:

1. **`skip-authors` glob expansion (`action.yml`):** the matching loop now runs with pathname expansion disabled — `set -f` before the loop and `set +f` after — so patterns like `alfresco-build*` are matched literally by `case` and are never expanded against the workspace.
2. **`min-words` validation (`action.yml`):** the value is now validated with `grep -Eq '^[0-9]+$'` before the numeric comparison and emits a clear `::error title=Invalid min-words` instead of a raw bash "integer expression expected".
3. **SHA pinning note (`docs/README.md`):** added a note in the `check-pr-description` section directing consumers to pin to a commit SHA per [Actions SHA pinning](#actions-sha-pinning), and that `min-words` must be a non-negative integer.
4. **CI coverage (`test.yml`):** added optional `pr-body`/`pr-author`/`pr-draft` override inputs and three steps in the `test` job — happy path, an asserted failure on a too-short body, and a bot-author skip. All logic paths were verified locally (happy/short/comment-only/invalid-min = fail-or-pass as expected; bot + `alfresco-build*` = skipped).

Resolving these conversations.

**Review by gionn — COMMENTED** [review]
Nice work overall — the gating logic and the meaningful-content parsing (stripping links/ticket keys/HTML comments) are solid.

One suggestion: consider refactoring the bash logic in `action.yml` out into a standalone script (e.g. `check-pr-description.sh`), and testing it with `bats` instead of the current end-to-end steps in `test.yml`. A few reasons:

- **Precedent in this repo**: several actions already follow this pattern — script + `tests/*.bats` — e.g. `dependabot-missing-actions-check/dependabot-coverage-check.sh` with `tests/dependabot-coverage-check.bats`, or `git-latest-tag`, `http-latency`, `send-teams-notification`. Keeping `check-pr-description` consistent makes it easier to navigate for anyone familiar with the other actions.
- **Faster, more isolated tests**: bats tests run as plain shell invocations against the script, no need to spin up a full `uses: ./.github/actions/...` composite step per case in CI. Easier to add many small cases (edge cases in the meaningful-text stripping, min-chars/min-words validation, skip-author/skip-branch globbing) without bloating `test.yml`.
- **Drop the test-only inputs**: `pr-body`, `pr-author`, `pr-draft`, and `pr-branch` are only there so `test.yml` can drive the action end-to-end — the descriptions themselves say "override mainly for testing." That's a bit of an action-shaped script. If the logic moves to a script that takes these as plain args/env vars, bats can call it directly with arbitrary inputs, and the composite action's inputs can be trimmed back to real configuration knobs (`min-chars`, `min-words`, `skip-authors`, `skip-branches`), with the action itself always sourcing PR body/author/draft/branch from the `github.event.pull_request` context rather than exposing test seams as public inputs.

Happy to help split this out if useful.

**Comment by GioLauria** [comment]
> Nice work overall — the gating logic and the meaningful-content parsing (stripping links/ticket keys/HTML comments) are solid.
> 
> One suggestion: consider refactoring the bash logic in `action.yml` out into a standalone script (e.g. `check-pr-description.sh`), and testing it with `bats` instead of the current end-to-end steps in `test.yml`. A few reasons:
> 
> * **Precedent in this repo**: several actions already follow this pattern — script + `tests/*.bats` — e.g. `dependabot-missing-actions-check/dependabot-coverage-check.sh` with `tests/dependabot-coverage-check.bats`, or `git-latest-tag`, `http-latency`, `send-teams-notification`. Keeping `check-pr-description` consistent makes it easier to navigate for anyone familiar with the other actions.
> * **Faster, more isolated tests**: bats tests run as plain shell invocations against the script, no need to spin up a full `uses: ./.github/actions/...` composite step per case in CI. Easier to add many small cases (edge cases in the meaningful-text stripping, min-chars/min-words validation, skip-author/skip-branch globbing) without bloating `test.yml`.
> * **Drop the test-only inputs**: `pr-body`, `pr-author`, `pr-draft`, and `pr-branch` are only there so `test.yml` can drive the action end-to-end — the descriptions themselves say "override mainly for testing." That's a bit of an action-shaped script. If the logic moves to a script that takes these as plain args/env vars, bats can call it directly with arbitrary inputs, and the composite action's inputs can be trimmed back to real configuration knobs (`min-chars`, `min-words`, `skip-authors`, `skip-branches`), with the action itself always sourcing PR body/author/draft/branch from the `github.event.pull_request` context rather than exposing test seams as public inputs.
> 
> Happy to help split this out if useful.

Adopted your suggestions

**Review by gionn — APPROVED** [review]

File: .github/actions/check-pr-description/action.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    default: "3"
+  skip-authors:
+    description: Space-separated shell glob patterns of PR author logins to skip, in addition to any '*[bot]' author.
+    required: false
+    default: "dependabot renovate alfresco-build*"
+  skip-branches:
+    description: Space-separated shell glob patterns of head branch names to skip (automated/bot PRs). Matched against github.head_ref.
+    required: false
+    default: "dependabot/* renovate/* updatecli* image-update* flux-* pr-* release-please--* changeset-release/* snyk-* whitesource* mend-* automated-* automation/*"
+
```
[the remark this answers is not in the document]
Comment (gionn) [reply .github/actions/check-pr-description/action.yml:22]: @GioLauria please bump the action description before merging

[the remark this answers is not in the document]
Comment (GioLauria) [reply .github/actions/check-pr-description/action.yml:22]: The PR description has been updated to match the implemented interface. The action exposes only `min-chars`, `min-words`, `skip-authors` and `skip-branches`. The `PR_BODY`/`PR_AUTHOR`/`PR_DRAFT`/`PR_BRANCH` values are read directly from `github.event.pull_request.*` / `github.head_ref` (not consumer-facing inputs); the bats tests exercise the script by setting those as env vars. No unsupported inputs are advertised anymore.

**Comment by atchertchian** [comment]
@GioLauria sorry do you mind changing action name for consistency with others? I suggest github-pr-check-description (since we have https://github.com/Alfresco/alfresco-build-tools/tree/master/.github/actions/github-pr-check-metadata)

**Comment by GioLauria** [comment]
> @GioLauria sorry do you mind changing action name for consistency with others? I suggest github-pr-check-description (since we have https://github.com/Alfresco/alfresco-build-tools/tree/master/.github/actions/github-pr-check-metadata)

I will pin it as a TO DO. Just merged it and all of the train PRs.
